Matchmaking queue ban.

y4ure opened the pull request.

y4ure commented on driftbase/api/matches.py:
```diff
@@ -886,6 +886,13 @@ def patch(self, args, match_id, player_id):
+            log_match_event(match_id, player_id,"gameserver.match.player_banned")
+            current_app.extensions["messagebus"].publish_message("match", {
+                "event": "match_player_banned", "match_id": match_id, "player_id": player_id
+            })
```
BattleServer will be updating the banned status of the player if
- In a ranked match
- Left the match and doesn't rejoin within rejoin time.
- Left the match and doesn't rejoin before the match ends.

y4ure commented on driftbase/api/matches.py:
```diff
@@ -586,7 +586,7 @@ def put(self, args, match_id):
                 if new_status == "started":
                     match.start_date = utcnow()
-                elif new_status == "completed":
+                elif new_status in ("completed", "ended"):
```
I see "end" in match active checks but is missing here.

Fixing battle server so aborted servers will have match_status set to "ended".

y4ure commented on driftbase/resources/flexmatch.py:
```diff
@@ -9,8 +9,12 @@
+    "ban_times": {
+        "time_tiers_minutes": [6, 15, 30, 60, 12 * 60],
+        "expire_time_minutes": 24 * 60
+    }
```
Escalating ban times, expires in an hour.

einarth reviewed (commented):
Don't see anything wrong with the code, but I feel that there should be a test for the ban mechanic itself and its escalation.

y4ure commented:
> Don't see anything wrong with the code, but I feel that there should be a test for the ban mechanic itself and its escalation.

Rewrote the logic to support separate banning for different matchmakers.
Updated unit test.

y4ure commented on driftbase/flexmatch.py:
```diff
@@ -793,3 +807,69 @@ def __init__(self, user_message, debug_info):
         self.debugs = debug_info
+
+class BanInfo(object):
+    test_store = None
```
Updated
